javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,60 @@
+let selects = document.querySelectorAll('select');
+let fromSelect = document.querySelector("#from-currency");
+let toSelect = document.querySelector("#to-currency");
+let inputAmount = document.querySelector("#amount");
+let convertedAmount = document.querySelector("#convertedAmount");
+
+
+// Fetch data from the API
+async function fetchData(urlAPI) {
+    try {
+        let response = await fetch(urlAPI);
+        let result = await response.json();
+        return result;
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+
+// Add options to select elements
+async function addSelectOptions() {
+    let data = await fetchData("https://open.er-api.com/v6/latest/USD");
+    for (let i = 0; i < selects.length; i++) {
+        for (const currency in data.rates) {
+            const option = document.createElement('option');
+            option.text = currency;
+            option.value = currency;
+            selects[i].add(option);
+        }
+    }
+    fromSelect.value = "EUR";
+}
+
+window.addEventListener('load', function () {
+    addSelectOptions();
+});
+
+
+// Add click eventlistener on button
+let convertButton = document.querySelector("#convert");
+convertButton.addEventListener("click", event => {
+    event.preventDefault()
+    getExchangeAmount();
+});
+
+async function getExchangeAmount() {
+        let data = await fetchData(`https://open.er-api.com/v6/latest/${fromSelect.value}`);
+
+        let exchangeRate = data.rates[toSelect.value];
+        let totalAmount = (inputAmount.value * exchangeRate).toFixed(2);
+        convertedAmount.innerHTML = `${inputAmount.value} ${fromSelect.value} = ${totalAmount} ${toSelect.value}`;
```

Comment:
Javascrcipt has a special API for formatinig numbers as currencies. Check it out https://gomakethings.com/how-to-format-a-number-as-currency-with-the-intl.numberformat-api-in-vanilla-javascript/